### PR TITLE
[EDA-1607] Add dumpfile for icarus

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
 - [x] To address an existing issue. If so, please provide a link to the issue: EDA-1607
 - [ ] Breaking new feature. If so, please describe details in the description part.

### Describe the technical details
Dump waveform file with user filename for icarus

 ### Which part of the code base require a change
 
 - [x] Library: simulation
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### CODE REVIEW

File: src/Simulation/Simulator.cpp [hunk 1]
```diff
@@ -735,6 +735,7 @@ std::string Simulator::SimulatorRunCommand(SimulationType simulation,
       if (m_waveType == WaveformType::FST) {
         command += " -fst";
       }
+      if (!m_waveFile.empty()) command += " -dumpfile=" + m_waveFile;
```
Comment (ravic-rs) [inline src/Simulation/Simulator.cpp:738]: What's the default name for wavefile in our GUI ? It appears to be "icarus" which doesn't align with other simulator defaults, correct ?

Comment (ravic-rs) [reply src/Simulation/Simulator.cpp:738]: Based off the settings, I think it should have been "sim_rtl.fst"
We only support fst for icarus via GUI as per the simulator docs.
image

Comment (KochynVolodymyr) [reply src/Simulation/Simulator.cpp:738]: this is issue with simulate command, I'll create another PR

Comment (KochynVolodymyr) [reply src/Simulation/Simulator.cpp:738]: Here is fix: https://github.com/os-fpga/FOEDAG/pull/1138

**Pull request merged**